presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -148,6 +149,27 @@ public void testTableWithNoSupportedColumns()
         }
     }
 
+    @Test
+    public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
+            throws Exception
+    {
+        try (AutoCloseable table = withTable("tpch.test_cleanup", "(column_a INTEGER)")) {
+            assertQuery("SHOW TABLES", "VALUES 'orders', 'test_cleanup'");
+
+            execute("ALTER TABLE tpch.test_cleanup ADD CHECK (column_a > 0)");
+
+            Exception thrown = null;
+            try {
+                assertUpdate("INSERT INTO tpch.test_cleanup(column_a) VALUES (0)", 1);
+            }
+            catch (Exception e) {
+                thrown = e;
+                assertQuery("SHOW TABLES", "VALUES 'orders', 'test_cleanup'");
```

Comment:
The test queries are run in multiple threads when using the Maven test runner, so this will fail due to other tests creating tables. Instead, you can use `SHOW TABLES LIKE 'tmp_presto_%'` and validate that nothing exists.